src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -0,0 +1,269 @@
+/* Copyright © 2018 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#include <toolkits/style_transfer/style_transfer.hpp>
+
+#include <map>
+#include <random>
+#include <string>
+
+#include <core/data/image/image_type.hpp>
+#include <model_server/lib/image_util.hpp>
+#include <model_server/lib/variant_deep_serialize.hpp>
+#include <toolkits/style_transfer/style_transfer_model_definition.hpp>
+
+namespace turi {
+namespace style_transfer {
+
+using turi::neural_net::compute_context;
+using turi::neural_net::float_array_map;
+using turi::neural_net::float_scalar;
+using turi::neural_net::shared_float_array;
+
+namespace {
+
+constexpr size_t STYLE_TRANSFER_VERSION = 1;
+
+constexpr int DEFAULT_BATCH_SIZE = 32;
+
+constexpr size_t MEMORY_REQUIRED_FOR_DEFAULT_BATCH_SIZE = 4294967296;
```

Comment:
Let's use the `lu` suffix here to ensure that we're defining a 64-bit constant. (The default is `int`, which on most of our platforms is 32-bit, which is not big enough to represent this number!)